fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -1420,12 +1435,15 @@ public PlanFragment visitPhysicalIcebergScan(OptExpression optExpression, ExecPl
                             .add(ScalarOperatorToExpr.buildExecExpression(predicate, formatterContext));
                 }
 
-                icebergScanNode.preProcessIcebergPredicate(node.getPredicate());
+                ScalarOperator icebergPredicate = !isEqDeleteScan ? node.getPredicate() :
+                        ((PhysicalIcebergEqualityDeleteScanOperator) node).getOriginPredicate();
```

Comment:
the schema of iceberg_equality_table is a subset of iceberg_table. eg: iceberg table schema: [c1, c2, c3]. the identifier column is c1. if the query predicate is `c1 > 1 and c2 < 3`, it can't be used as a predicate of equality_table. so we named it originPredicate in the equality_table.